runtime/bin/directory_linux.cc

```diff
@@ -277,28 +275,27 @@ static bool DeleteRecursively(PathBuffer* path) {
 
   // Iterate the directory and delete all files and directories.
   int path_length = path->length();
-  dirent entry;
   dirent* result;
-  while (NO_RETRY_EXPECTED(readdir_r(dir_pointer, &entry, &result)) == 0) {
+  while ((result = readdir(dir_pointer)) != NULL) {
```

Comment:
This doesn't do the same thing anymore: it doesn't closedir and remove since it will never enter line 282.

What about:
```
while (true) {
  dirent* result = readdir(dir_pointer);
  if (result == NULL) { ...
```